### PR TITLE
Add natspec comments to contracts

### PR DESCRIPTION
**Pull request by maurelian**
This PR add natspec conformant documentation comments to all contracts. This will enable automated generation of the basic contract documentation.

### CODE REVIEW

**Comment by maraoz** [comment]
This is amazing work!!! Thanks @maurelian and @JGcarv! 
Can you check the compiler errors we're getting before we review?

**Comment by JGcarv** [comment]
I'll do that later today!

**Comment by maraoz** [comment]
Reviewing...

File: contracts/token/MintableToken.sol [hunk 1]
```diff
@@ -23,18 +20,29 @@ contract MintableToken is StandardToken, Ownable {
...
+   */
   function mint(address _to, uint _amount) onlyOwner canMint returns (bool) {
     totalSupply = totalSupply.add(_amount);
     balances[_to] = balances[_to].add(_amount);
     Mint(_to, _amount);
     return true;
   }
 
+  /**
+   * @dev Function to spot minting new tokens.
```
Comment (maraoz) [inline contracts/token/MintableToken.sol]: stop

File: contracts/token/PausableToken.sol [hunk 2]
```diff
@@ -15,11 +15,22 @@ import '../lifecycle/Pausable.sol';
 
 contract PausableToken is Pausable, StandardToken {
 
+  /**
+  * @dev Transfer from one addres to another.
+  * @param _to address The addres to transfer to
+  * @param _value uint The value to transfer
+  */
```
Comment (maraoz) [inline contracts/token/PausableToken.sol]: Do we really need to repeat the docs from StandardToken here? Can't we say something like `@see StandardToken#transfer` ?

File: contracts/token/PausableToken.sol [hunk 3]
```diff
@@ -15,11 +15,22 @@ import '../lifecycle/Pausable.sol';
...
   function transfer(address _to, uint _value) whenNotPaused {
     return super.transfer(_to, _value);
   }
 
+  /**
+  * @dev Transfer from one addres to another.
+  * @param _to address The addres to transfer to
+  * @param _from address The addres to transfer from
+  * @param _value uint The value to transfer
+  */
```
Comment (maraoz) [inline contracts/token/PausableToken.sol]: same as above

File: contracts/token/VestedToken.sol [hunk 4]
```diff
@@ -15,6 +20,14 @@ contract VestedToken is StandardToken, LimitedTransferToken {
...
   mapping (address => TokenGrant[]) public grants;
 
+  /**
+   * @dev Grant tokens to a specified address
+   * @param _to address The address which the tokens will be granted to.
+   * @param _value uint256 The amount of tokens to be granted.
+   * @param _start uint64 Represents time of the begining of the grant.
+   * @param _cliff uint64 Represents the cliff period.
+   * @param _vesting uint64 Represents the vesting period.
+   */
```
Comment (maraoz) [inline contracts/token/VestedToken.sol]: I'd remove all the "Represents" and just have:
```
@param _start uint64 Time of the beginning of the grant.
```

etc.

File: contracts/token/VestedToken.sol [hunk 5]
```diff
@@ -15,6 +20,14 @@ contract VestedToken is StandardToken, LimitedTransferToken {
 
   mapping (address => TokenGrant[]) public grants;
 
+  /**
+   * @dev Grant tokens to a specified address
+   * @param _to address The address which the tokens will be granted to.
+   * @param _value uint256 The amount of tokens to be granted.
+   * @param _start uint64 Represents time of the begining of the grant.
```
Comment (maraoz) [inline contracts/token/VestedToken.sol]: beginning

File: contracts/token/VestedToken.sol [hunk 6]
```diff
@@ -83,6 +120,15 @@ contract VestedToken is StandardToken, LimitedTransferToken {
     );
   }
 
+  /**
+  * @dev Calculate amount of vested tokens at a specifc time.
+  * @param tokens uint256 The amount of tokens grantted.
+  * @param time uint64 The time to be checked
+  * @param start uint64 A time representing the begining of the grant
```
Comment (maraoz) [inline contracts/token/VestedToken.sol]: beginning

File: contracts/token/VestedToken.sol [hunk 7]
```diff
@@ -57,10 +76,22 @@ contract VestedToken is StandardToken, LimitedTransferToken {
...
+   */
   function tokenGrantsCount(address _holder) constant returns (uint index) {
     return grants[_holder].length;
   }
 
+  /**
+   * @dev Get all information about a specifc grant.
+   * @param _holder The address which will have its tokens revoked.
+   * @param _grantId The id of the token grant.
+   * @return Returns all the values that represent a TokenGrant(address, value, start, cliff
```
Comment (maraoz) [inline contracts/token/VestedToken.sol]: returns all the information about a `TokenGrant`

**Comment by maurelian** [comment]
@maraoz rebased from upstream master

**Comment by maraoz** [comment]
Amazing work!!! :confetti_ball: :fireworks:

**Comment by maurelian** [comment]
Yeah, there's still work to do. For example, not all params have a type specified. And not the format of the `@returns` statements is somewhat varied. 

Also assessing your suggestion to not duplicate documentation by using `@see`. 

BTW, I discovered that you can parse the comments to JSON using `solc --devdoc`, so that's a great start for automatically generating HTML docs.

**Pull request merged**